backend/api/services/ComplianceReportService.py

```diff
@@ -196,6 +196,16 @@ def create_history(compliance_report, is_new=False):
             else:
                 role_id = user.roles.first().id
 
+        history = ComplianceReportHistory.objects.filter(
+            Q(compliance_report_id=compliance_report.id) &
+            Q(status__fuel_supplier_status__status=compliance_report.status.fuel_supplier_status_id) &
+            Q(status__analyst_status__status=compliance_report.status.analyst_status_id) &
+            Q(status__manager_status__status=compliance_report.status.manager_status_id) &
+            Q(status__director_status__status=compliance_report.status.director_status_id))
+        # if a history record is already present with same status then don't create a new one.
+        if len(history) > 0:
```

Comment:
This is good, but I think we should address the problem on the frontend as well. The button should be disabled after click, until a successful or failed response comes back. The view I think reloads anyway when the api call finishes so a disable should work.